web-ng/src/app/components/feature-details/feature-details.component.ts

```diff
@@ -0,0 +1,126 @@
+import { Layer } from '../../shared/models/layer.model';
+import { getPinImageSource } from '../map/ground-pin';
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
+import { ConfirmationDialogComponent } from '../confirmation-dialog/confirmation-dialog.component';
+import { MatDialog } from '@angular/material/dialog';
+import { Router } from '@angular/router';
+import { DataStoreService } from '../../services/data-store/data-store.service';
+import { NavigationService } from './../../services/router/router.service';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-feature-details',
+  templateUrl: './feature-details.component.html',
+  styleUrls: ['./feature-details.component.scss'],
+})
+export class FeatureDetailsComponent implements OnInit, OnDestroy {
+  @Input() layer?: Layer;
+  @Input() actionsType: LayerListItemActionsType =
+    LayerListItemActionsType.MENU;
+  projectId?: string | null;
+  featureId?: string | null;
+  layerPinUrl: SafeUrl;
+  readonly lang: string;
+  readonly layerListItemActionsType = LayerListItemActionsType;
+  subscription: Subscription = new Subscription();
+
+  constructor(
+    private sanitizer: DomSanitizer,
+    private confirmationDialog: MatDialog,
+    private router: Router,
+    private dataStoreService: DataStoreService,
+    private navigationService: NavigationService,
+    private zone: NgZone
+  ) {
+    // TODO: Make dynamic to support i18n.
+    this.lang = 'en';
+    this.layerPinUrl = sanitizer.bypassSecurityTrustUrl(getPinImageSource());
+  }
+
+  ngOnInit() {
+    this.layerPinUrl = this.sanitizer.bypassSecurityTrustUrl(
+      getPinImageSource(this.layer?.color)
+    );
+    this.subscription.add(
+      this.navigationService.getFeatureId$().subscribe(id => {
+        this.featureId = id;
+      })
+    );
+    this.subscription.add(
+      this.navigationService.getProjectId$().subscribe(id => {
+        this.projectId = id;
+      })
+    );
+  }
+
+  ngOnChanges() {
+    this.layerPinUrl = this.sanitizer.bypassSecurityTrustUrl(
+      getPinImageSource(this.layer?.color)
+    );
+  }
+
+  onGoBackClick() {
+    this.navigationService.setFeatureId(null);
+  }
+
+  onDeleteFeature() {
+    const dialogRef = this.confirmationDialog.open(
+      ConfirmationDialogComponent,
+      {
+        maxWidth: '500px',
+        data: {
+          title: 'Warning',
+          message:
+            'Are you sure you wish to delete this feature? Any associated data including all observations in this feature will be lost. This cannot be undone.',
+        },
+      }
+    );
+
+    dialogRef.afterClosed().subscribe(async dialogResult => {
+      if (dialogResult) {
+        await this.deleteFeature();
+      }
+    });
+  }
+
+  async deleteFeature() {
+    if (!this.projectId || !this.featureId) {
+      return;
+    }
+    await this.dataStoreService.deleteFeature(this.projectId, this.featureId);
```

Comment:
If you add this functionality please also make sure deleteFeature() deletes observations.